src/app/(private)/emergency-evacuation-applications/[id]/page.tsx

```diff
@@ -0,0 +1,466 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import { Input } from '@/components/ui/input'
+import { formatDateTime } from '@/lib/formatDateTime'
+import {
+  FormItem,
+  FormField,
+  FormControl,
+  FormLabel,
+  Form,
+} from '@/components/ui/form'
+import { useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card'
+import { useTranslation } from 'react-i18next'
+import { LoadingSpinner } from '@/components/ui/loadingSpinner'
+import { useToast } from '@/components/ui/use-toast'
+import { formatPhoneNumber } from '@/lib/formatPhoneNumber'
+import PrivateRoute from '@/app/hocs/isAuth'
+import { Permission } from '@/constants/permissions'
+import { FormSchema } from '@/modules/emergencyEvacuationApplications/constants/formSchema'
+import { EmergencyEvacuationApplication } from '@/modules/emergencyEvacuationApplications/constants/types'
+import { getEmergencyEvacuationApplication } from '@/modules/emergencyEvacuationApplications/service'
+import { Checkbox } from '@/components/ui/checkbox'
+import { getStatusLabel } from '@/modules/emergencyEvacuationApplications/components/status'
+
+const Page = ({ params }: { params: { slug: string; id: string } }) => {
+  const { t } = useTranslation()
+  const { toast } = useToast()
+  const form = useForm({
+    resolver: zodResolver(FormSchema),
+  })
+  const { control } = form
+
+  const [
+    emergencyEvacuationApplicationDetails,
+    setEmergencyEvacuationApplicationDetails,
+  ] = useState<EmergencyEvacuationApplication | null>(null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => {
+    const fetchDetails = () => {
+      getEmergencyEvacuationApplication(params.id)
+        .then((response) => {
+          if (response.data.isSuccess) {
```

Comment:
then içinde if else bloğunu neden var, thene zaten başarılı olursa giriyor. BE den errorlar doğru döndüğü için direk catch içine düşüyor.

Reply:
buradaki ekstra check good/best practice olarak geçiyor. HTTP request success olunca then'e giriliyor, burada isSuccess'i de check etmekte fayda var diye düşünüyorum.

Reply:
zaten success olunca buraya giriyor senin de dediğin gibi, burada extra check sadece kodu kalabalıklaştırmış

Reply:
HTTP request OK olup da girebilir ama data da bir eksiklik olabilir backend o zaman fail ile gönderebilir ama gene de http requestde sorun olmadığı için then'e girer diye biliyorum. o yüzden bu check konuluyor. değiştirebilirim son kararına göre.

Reply:
Bazı yerlerde BE kötü yazılınca böyle bir stepe ihtiyaç duyulabilir, ama buradaki BE için öyle bir ihtiyacımız yok.

Reply:
tamamdır, BE bilmediğim için öyle bi check koymuştum, düzenliyorum🐰

Reply:
@kevseroz diğer sayfalarda da bunun gözden kaçtığı yerler var sanırım, onları da düzenleyeyim mi yoksa zaten mergelendikleri için issue filan mı açılır?

Reply:
İstersen issue açalım öyle devam et daha düzenli olsun

Reply:
değiştirdim